### DISCUSSION TITLE
MathJax formula into SVG, how-to?

### DISCUSSION THREAD

**Discussion by masa300V** [asciidocfx/AsciidocFX#655] [Q&A] [answered]
Hi!

May I know how I can save my MathJax formula into SVG?

When I specify a target file below, it generates test.svg.png...
`[latexmath, file="test.svg"]`

![image](https://github.com/user-attachments/assets/1e543979-3fed-4033-b5fa-276d7064ec4a)

masashi

**Comment by rahmanusta** [comment, marked as answer]
Hi, not supported anymore, only png output is supported.

**Reply by masa300V** [reply]
I see, thanks!

**One more question (below)**, based on my half-cooked hacking.

By using MathJax 3 or Mathjax 4, I see that I can generate a SVG inside JS of the "conf/public/mathajx.html" with a small tweak.

***

Step-1: Use MathJax 4 with afx.snapshotFormula(), only key-change shown here, this is fully working on my temporary hack and I can use many fonts with MathJax 4.
````
96,97c87
<                     MathJax.Hub.Queue(["Typeset", MathJax.Hub], function () {
<                         try {
---
>                     MathJax.typesetPromise().then(() => {
````

***

step-2: Generate SVG with MathJax 4, this is not fully working...yet...
````
<                     MathJax.typesetPromise().then(() => {
---
>                     MathJax.tex2svgPromise(popped.formula).then(() => {
````

At this point, inside JS of this "conf/public/mathjax.html", we do have a SVG data...


**Q:**  Is there any I/O I/F we can save SVG data into a local drive specified in the "popped.imageInfo"?


Right now, my generated SVG-base formula is saved as "snapshot" by "afx.snapshotFormula"...I want to have a method like "afx.saveSVGFomula"...where I don't have skillset...

**Reply by masa300V** [reply]
This is a patch for my "step-1".

Right now, I'm using MathJax 4 on the server but can use local JS easily.

[MJ4.patch](https://github.com/user-attachments/files/16573729/MJ4.patch)

**Comment by rahmanusta** [comment]
I know but somehow I was dropped that feature. The reason might be the JavaFX webview is only partially supporting the svg format that the webview shows some of svg images broken. I can put the code back if you create a pull request.

**Reply by masa300V** [reply]
Thanks!

What I'm thinking is to use "conf/public/mathjax.html" (of MathJax) directly generate "SVG data" and save it without showing SVG on the webview (thus webview rendering is not an issue).  I'm not good at JavaFX webview side...and not sure the detail of your "the code back" (as I started using the software after loosing SVG support).

To me, I want to access an I/F of something like  afx.saveSVGImage() inside the "conf/pub/mathjax.html" JS, instead of afx.snapshotFormula().  Inside the JS of conf/pub/mathjax.html, I can generate SVG data-byte and I want this afx.saveSVGImage() to save it on the cache or into a local file per "file=..." option.  It the code you can place back does this?
